libraries/standard/http/src/http_client.c

```diff
@@ -3,59 +3,75 @@
+static HTTPStatus_t sendHttpData( const HTTPTransportInterface_t * pTransport,
+                                  const uint8_t * pBuffer,
+                                  size_t bufferLen );
 
 /**
  * @brief Send the HTTP headers over the transport send interface.
  *
  * @param[in] pTransport Transport interface.
  * @param[in] pRequestHeaders Request headers to send, it includes the buffer
  * and length.
+ * @param[in] reqBodyBufLen The length of the request body to be sent. This is
+ * used to generated a Content-Length header.
+ * @param[in] flags Application provided flags to #HTTPClient_Send.
  *
  * @return #HTTP_SUCCESS if successful. If there was a network error or less
- * bytes than what was specified were sent, then #HTTP_NETWORK_ERROR is returned.
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
  */
 static HTTPStatus_t sendHttpHeaders( const HTTPTransportInterface_t * pTransport,
-                                     const HTTPRequestHeaders_t * pRequestHeaders );
+                                     const HTTPRequestHeaders_t * pRequestHeaders,
+                                     size_t reqBodyBufLen,
+                                     uint32_t flags );
+
+/**
+ * @brief Send the HTTP Content-Length header line over the transport interface.
+ *
+ * @param[in] pTransport Transport interface.
+ * @param[in] contentLength The Content-Length header value to send.
+ *
+ * @return #HTTP_SUCCESS if successful. If there was a network error or less
+ * bytes than what were specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
+ */
+static HTTPStatus_t sendContentLength( const HTTPTransportInterface_t * pTransport,
+                                       size_t contentLength );
 
 /**
  * @brief Send the HTTP body over the transport send interface.
  *
  * @param[in] pTransport Transport interface.
  * @param[in] pRequestBodyBuf Request body buffer.
- * @param[in] reqBodyLen Length of the request body buffer.
+ * @param[in] reqBodyBufLen Length of the request body buffer.
  *
  * @return #HTTP_SUCCESS if successful. If there was a network error or less
- * bytes than what was specified were sent, then #HTTP_NETWORK_ERROR is returned.
+ * bytes than what was specified were sent, then #HTTP_NETWORK_ERROR is
+ * returned.
  */
 static HTTPStatus_t sendHttpBody( const HTTPTransportInterface_t * pTransport,
                                   const uint8_t * pRequestBodyBuf,
                                   size_t reqBodyBufLen );
 
 /**
- * @brief Write header based on parameters. This method also adds a trailing "\r\n".
- * If a trailing "\r\n" already exists in the HTTP header, this method backtracks
- * in order to write over it and updates the length accordingly.
+ * @brief Write header based on parameters. This method also adds a trailing
+ * "\r\n". If a trailing "\r\n" already exists in the HTTP header, this method
```

Comment:
Funny how we have different preferences when it comes to organizing length of multi-line comments. One thing I wish `uncrustify` could do automatically lol.